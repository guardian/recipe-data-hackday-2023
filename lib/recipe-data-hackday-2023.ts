import type { GuStackProps } from '@guardian/cdk/lib/constructs/core';
import { GuStack } from '@guardian/cdk/lib/constructs/core';
import { Tags } from 'aws-cdk-lib';
import type { App } from 'aws-cdk-lib';
import { aws_s3 as s3 } from 'aws-cdk-lib';
import { AttributeType, Table } from 'aws-cdk-lib/aws-dynamodb';

export class RecipeDataHackday2023 extends GuStack {
	constructor(scope: App, id: string, props: GuStackProps) {
		super(scope, id, props);

		new Table(this, 'RecipeTable', {
			tableName: 'recipes',
			partitionKey: { name: 'path', type: AttributeType.STRING },
			sortKey: { name: 'recipeId', type: AttributeType.NUMBER },
		});

		new Table(this, 'RecipeEditsTable', {
			tableName: 'recipes-edited',
			partitionKey: { name: 'path', type: AttributeType.STRING },
			sortKey: { name: 'recipeId', type: AttributeType.NUMBER },
		});

		new s3.Bucket(this, 'StructuredRecipes'); 

		Tags.of(this).add('App', 'recipe-data-hackday-2023');
	}
}
