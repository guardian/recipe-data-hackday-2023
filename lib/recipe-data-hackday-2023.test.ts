import { App } from "aws-cdk-lib";
import { Template } from "aws-cdk-lib/assertions";
import { RecipeDataHackday2023 } from "./recipe-data-hackday-2023";

describe("The RecipeDataHackday2023 stack", () => {
  it("matches the snapshot", () => {
    const app = new App();
    const stack = new RecipeDataHackday2023(app, "RecipeDataHackday2023", { stack: "playground", stage: "TEST" });
    const template = Template.fromStack(stack);
    expect(template.toJSON()).toMatchSnapshot();
  });
});
