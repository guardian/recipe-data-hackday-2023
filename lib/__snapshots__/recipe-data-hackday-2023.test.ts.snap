// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The RecipeDataHackday2023 stack matches the snapshot 1`] = `
{
  "Metadata": {
    "gu:cdk:constructs": [],
    "gu:cdk:version": "TEST",
  },
  "Resources": {
    "RecipeEditsTableEDB53D3E": {
      "DeletionPolicy": "Retain",
      "Properties": {
        "AttributeDefinitions": [
          {
            "AttributeName": "path",
            "AttributeType": "S",
          },
          {
            "AttributeName": "recipeId",
            "AttributeType": "N",
          },
        ],
        "KeySchema": [
          {
            "AttributeName": "path",
            "KeyType": "HASH",
          },
          {
            "AttributeName": "recipeId",
            "KeyType": "RANGE",
          },
        ],
        "ProvisionedThroughput": {
          "ReadCapacityUnits": 5,
          "WriteCapacityUnits": 5,
        },
        "TableName": "recipes-edited",
        "Tags": [
          {
            "Key": "App",
            "Value": "recipe-data-hackday-2023",
          },
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/recipe-data-hackday-2023",
          },
          {
            "Key": "Stack",
            "Value": "playground",
          },
          {
            "Key": "Stage",
            "Value": "TEST",
          },
        ],
      },
      "Type": "AWS::DynamoDB::Table",
      "UpdateReplacePolicy": "Retain",
    },
    "RecipeTable9AD52CED": {
      "DeletionPolicy": "Retain",
      "Properties": {
        "AttributeDefinitions": [
          {
            "AttributeName": "path",
            "AttributeType": "S",
          },
          {
            "AttributeName": "recipeId",
            "AttributeType": "N",
          },
        ],
        "KeySchema": [
          {
            "AttributeName": "path",
            "KeyType": "HASH",
          },
          {
            "AttributeName": "recipeId",
            "KeyType": "RANGE",
          },
        ],
        "ProvisionedThroughput": {
          "ReadCapacityUnits": 5,
          "WriteCapacityUnits": 5,
        },
        "TableName": "recipes",
        "Tags": [
          {
            "Key": "App",
            "Value": "recipe-data-hackday-2023",
          },
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/recipe-data-hackday-2023",
          },
          {
            "Key": "Stack",
            "Value": "playground",
          },
          {
            "Key": "Stage",
            "Value": "TEST",
          },
        ],
      },
      "Type": "AWS::DynamoDB::Table",
      "UpdateReplacePolicy": "Retain",
    },
    "StructuredRecipesFF7C8293": {
      "DeletionPolicy": "Retain",
      "Properties": {
        "BucketName": "recipe-data-hackday-2023",
        "Tags": [
          {
            "Key": "App",
            "Value": "recipe-data-hackday-2023",
          },
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/recipe-data-hackday-2023",
          },
          {
            "Key": "Stack",
            "Value": "playground",
          },
          {
            "Key": "Stage",
            "Value": "TEST",
          },
        ],
      },
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Retain",
    },
  },
}
`;
